src/remote/entrypoints.py

```diff
@@ -179,16 +181,41 @@ def remote_set(index: int):
 @click.option("-m", "--mirror", is_flag=True, help="mirror local files on the remote host")
 @click.option("-v", "--verbose", is_flag=True, help="increase verbosity")
 @click.option("-e", is_flag=True, help="(deprecated) kept for backward compatibility, noop")
+@click.option(
+    "-t",
+    "--tunnel",
+    "port_args",
+    type=str,
+    help="Enable local port forwarding. Pass value as <remote port>:<local port>. \
+If local port is not passed, the local port value would be set to <remote port> value by default",
+)
 @click.argument("command", nargs=-1, required=True)
 @log_exceptions
-def remote(command: List[str], dry_run: bool, mirror: bool, verbose: bool, e: bool):
+def remote(command: List[str], dry_run: bool, mirror: bool, verbose: bool, e: bool, port_args: Optional[str]):
     """Sync local workspace files to remote machine, execute the COMMAND and sync files back regardless of the result"""
 
     if verbose:
         logging.basicConfig(level=logging.INFO, format=BASE_LOGGING_FORMAT)
 
+    port_forwarding_config = None
+    if port_args:
+        ports: List = port_args.split(":")
+        if len(ports) > 2:
+            click.secho("Please pass a valid value to enable local port forwarding", fg="yellow")
+            sys.exit(1)
+        try:
+            if len(ports) == 1:
+                port_forwarding_config = PortForwardingConfig(remote_port=int(ports[0]), local_port=int(ports[0]))
+            else:
+                port_forwarding_config = PortForwardingConfig(remote_port=int(ports[0]), local_port=int(ports[1]))
+        except ValueError:
+            click.secho("Please pass valid integer value for ports", fg="yellow")
+            sys.exit(1)
```

Comment:
I would make sense to create a separate function in util for it, because you will definitely reuse this code later for the internal implementation